--- wide-dhcpv6-20080615/dhcp6c.c	2011-12-23 13:04:13.117985411 +0800
+++ wide-dhcpv6-20080615.new/dhcp6c.c	2011-12-23 13:07:41.512120000 +0800
@@ -163,6 +163,7 @@
 	char *progname;
 	FILE *pidfp;
 	struct dhcp6_if *ifp;
+	int fd;
 
 #ifndef HAVE_ARC4RANDOM
 	srandom(time(NULL) & getpid());
@@ -223,8 +224,11 @@
 		exit(0);
 	}
 
-	if (foreground == 0)
+	if (foreground == 0) {
 		openlog(progname, LOG_NDELAY|LOG_PID, LOG_DAEMON);
+		for (fd = 3; fd < 1024; fd++)
+			close(fd);
+	}
 
 	setloglevel(debug);
 
@@ -253,6 +257,10 @@
 	if ((pidfp = fopen(pid_file, "w")) != NULL) {
 		fprintf(pidfp, "%d\n", pid);
 		fclose(pidfp);
+	} else { 
+	    fprintf(stderr, "Unable to write to %s: %s\n", pid_file, strerror(errno));
+	    fflush(stderr);
+	    abort(); 
 	}
 
 	client6_startall(0);
@@ -780,6 +788,12 @@
 
 	switch(command) {
 	case DHCP6CTL_COMMAND_START:
+		/* the ifid might have changed, so reset it before releasing the lease */
+		if (ifreset(ifp)) {
+			dprintf(LOG_NOTICE, FNAME, "failed to reset %s",
+				ifname);
+			return (-1);
+		}	    
 		free_resources(ifp);
 		if (client6_start(ifp)) {
 			ddprintf(LOG_NOTICE, FNAME, "failed to restart %s",
