--- wide-dhcpv6-20080615/common.c	2008-06-15 13:18:41.000000000 +0530
+++ wide-dhcpv6-20080615_mod/common.c	2010-03-30 21:40:12.000000000 +0530
@@ -995,7 +995,7 @@
 {
 	FILE *fp = NULL;
 	u_int16_t len = 0, hwtype;
-	struct dhcp6opt_duid_type1 *dp; /* we only support the type1 DUID */
+	struct dhcp6opt_duid_type3 *dp; /* we only support the type1 DUID */
 	char tmpbuf[256];	/* DUID should be no more than 256 bytes */
 
 	if ((fp = fopen(idfile, "r")) == NULL && errno != ENOENT)
@@ -1016,7 +1016,7 @@
 			    "failed to get a hardware address");
 			goto fail;
 		}
-		len = l + sizeof(struct dhcp6opt_duid_type1);
+		len = l + sizeof(struct dhcp6opt_duid_type3);
 	}
 
 	memset(duid, 0, sizeof(*duid));
@@ -1039,12 +1039,12 @@
 	} else {
 		u_int64_t t64;
 
-		dp = (struct dhcp6opt_duid_type1 *)duid->duid_id;
-		dp->dh6_duid1_type = htons(1); /* type 1 */
+		dp = (struct dhcp6opt_duid_type3 *)duid->duid_id;
+		dp->dh6_duid1_type = htons(3); /* type 3 */
 		dp->dh6_duid1_hwtype = htons(hwtype);
 		/* time is Jan 1, 2000 (UTC), modulo 2^32 */
-		t64 = (u_int64_t)(time(NULL) - 946684800);
-		dp->dh6_duid1_time = htonl((u_long)(t64 & 0xffffffff));
+		//t64 = (u_int64_t)(time(NULL) - 946684800);
+		//dp->dh6_duid1_time = htonl((u_long)(t64 & 0xffffffff));
 		memcpy((void *)(dp + 1), tmpbuf, (len - sizeof(*dp)));
 
 		dprintf(LOG_DEBUG, FNAME, "generated a new DUID: %s",
